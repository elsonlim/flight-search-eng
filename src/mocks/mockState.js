import moment from 'moment';

export const flights = [
    {
        "id": 0,
        "IATA": "DL",
        "flightNo": 90,
        "from": "USA",
        "to": "HKG",
        "price": {"amount": 261, "currency": "USD"},
        "time": {"arrival": 609, "departure": 425, "duration": 104}
    }, {
        "id": 1,
        "IATA": "DL",
        "flightNo": 91,
        "from": "HKG",
        "to": "USA",
        "price": {"amount": 232, "currency": "USD"},
        "time": {"arrival": 1956, "departure": 1730, "duration": 146}
    },
];

export const searchParams = {
    isReturn: true,
    from: 'USA',
    to: 'HKG',
    depatureDate: moment(new Date()),
    arrivalDate: moment(new Date()),
    numberOfPassengers: 1,
}

export const priceRange = {
    filter: {min: 0, max: 2000},
    search: {min: 200, max: 1000},
}

export default {
    flights,
    searchType: false,
    searchParams,
    priceRange
};

    