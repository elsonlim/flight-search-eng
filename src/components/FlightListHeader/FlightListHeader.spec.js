import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import enzyme, { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import { FlightListHeader } from './index';
import mockState from '../../mocks/mockState';

enzyme.configure({ adapter: new Adapter() });

describe('FlightListHeader', () => {

    let props;
    let wrapper;
    let store;
    const mockStore = configureStore();

    beforeEach(() => {
        props = mockState;
        store =  mockStore(mockState);
        wrapper = shallow(<FlightListHeader {...props} />);
    });
    
    describe('snapshot test', () => {
		test('show FlightListHeader', () => {
			const tree = renderer.create(<Provider store={store}><FlightListHeader {...props} /></Provider>).toJSON();
			expect(tree).toMatchSnapshot();
		});
    });

    describe('getFlightDate', () => {
        test('show 1 date field if is not return flight', () => {
            expect(wrapper.instance().getFlightDate(false)).toHaveLength(1);
        });

        test('show 2 date field if is not return flight', () => {
            expect(wrapper.instance().getFlightDate(true)).toHaveLength(2);
        });
    });
});
