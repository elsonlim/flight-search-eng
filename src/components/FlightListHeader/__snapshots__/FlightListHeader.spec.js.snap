// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightList snapshot test show FlightListHeader 1`] = `
<div
  className="flight-list-header"
>
  <div
    className="results-title"
  >
    Your Results
  </div>
  <div
    className="results-date"
  >
    <div>
      Depart: 
      2018-5-28
    </div>
    <div>
      Return: 
      2018-5-28
    </div>
  </div>
</div>
`;

exports[`FlightListHeader snapshot test show FlightListHeader 1`] = `
<div
  className="flight-list-header"
>
  <div
    className="results-title"
  >
    Your Results
  </div>
  <div
    className="results-date"
  >
    <div>
      Depart: 
      2018-5-28
    </div>
    <div>
      Return: 
      2018-5-28
    </div>
  </div>
</div>
`;
